### PR TITLE
Code Config Docs

### PR DESCRIPTION
**Pull request by seancdavis**
Lots going on here.

**Cleaning Up Frontmatter**

There are so many files changes because I removed the `section` property from every doc, which we can safely do now. (The breadcrumbs rely on _either_ the URL or the physical files in the path tree. And the nav is manually declared, separate from file structure.)

**Nav Updates**

Because of the structural change (mentioned below) for config docs, I've adjusted the nav's active state to look only for a partial match. That means when you're in a child page of a section, both the section and the child will be highlighted. I suspect we'll iterate on that when addressing the growing nav and introducing accordions or some other vertical space-saving measure.

**Configuration Overviews**

- The [main config page](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config) is now a brief overview that nudges the reader toward the code config page, but also mentions that we have enterprise guides.
    - **What I really want here is a method for pointing readers toward signing up for enterprise**. I'm going to bring that up for discussion with the group.
- The [Code Config page](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/code-config) is now rearranged.
    - I pulled in some content that existed elsewhere and played around with the order of things.
    - There's a lot going on here, so I [**added a story for tracking parts of and progress within a docs page**](https://www.pivotaltracker.com/story/show/176896646).
- There is [a new UI Config page](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/ui-config).
    - It talks about the enterprise edition. Most of what I did here, I pulled from the [First Time Setup page](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/getting-started/first-time-setup), which will likely become focused on Code Config when [the first five minutes video story](https://www.pivotaltracker.com/story/show/176808051) is complete.
    - Again, here, we're missing a CTA for the enterprise edition (which should probably be on every page that mentions enterprise). [I made a story](https://www.pivotaltracker.com/story/show/176902057) for it.

**Config Docs**

Where we previously had UI Guides, we now have Config Guides for both the Community Edition and Enterprise Edition. I spiked being clever about this navigation, but didn't get too far. I think it's worthwhile to spend more time with it, but for now, the base config guide route redirects to the community version, and links to the enterprise versions are added manually on a few different pages.

Example: `/docs/config/apps` redirects to `/docs/config/apps/community`, And both `/docs/config/apps/community` and `/docs/config/apps/enterprise` open with links to each other.

The Config Guides are:

- [Adding Apps](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/apps/community)
- [Adding Apps (Enterprise)](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/apps/enterprise)
- [Adding Sources](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/sources/community)
- [Adding Sources (Enterprise)](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/sources/enterprise)
- [Adding Properties to Sources](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/properties/community)
- [Adding Properties to Sources (Enterprise)](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/properties/enterprise)
- [Creating Groups](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/groups/community)
- [Creating Groups (Enterprise)](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/groups/enterprise)
- [Adding Destinations](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/destinations/community)
- [Adding Destinations (Enterprise)](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/destinations/enterprise)
- [Settings](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/config/settings)

**Config Support**

After going through these topics, it felt like [Environment](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/support/environment), [Permissions](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/support/permissions), and [Secrets](https://www-grouparoo-com-git-176694421-code-config-docs.grouparoo.vercel.app/docs/support/secrets) were more of support articles, rather than references that helped the reader configure their Grouparoo application. Therefore, I moved those to the Support section and added redirects where appropriate.

**Discrepancy in Steps**

While I was wrapping up this work, it occurred to me that there is a discrepancy in the order of tasks, depending on which Edition you're using. If we're using Community and relying on Code Config (which is the recommendation), then everything is fine. But if we're using the Enterprise Edition, and if we want to configure it using the UI, we _technically_ have to run the application before we configure it. I don't think it's a huge deal or anything we have to change right now, it just feels a little funny.

I didn't really notice it until putting the Settings page together, since the recommendation with Settings is that they are configured through the UI. And that means we do have to run the application before you can use this guide.

I looked back at our early iterations on the taxonomy and we had shoved Enterprise Guides underneath Running Grouparoo. Maybe all those UI Guides should move under the Run section. Thoughts?

### CODE REVIEW

File: pages/docs/config.mdx [hunk 1]
```diff
@@ -1,56 +1,41 @@
...
 
-## Code Config
-
-To use Grouparoo to import and export your data, you will be configuring our main objects: Apps, Sources, Destinations, Properties, etc. You can [learn more about Grouparoo's product concepts here](/docs/getting-started/product-concepts). You will be using the Grouparoo CLI's `generate`, `validate`, and `apply` commands to ensure that the application can read from your Sources and write to your Destinations.
-
-Configuration files are either JSON or Javscript which allow you the flexibility to manipulate data or load configuration from external sources:
-
-```js
-// file: config/apps/data_warehouse.js
+<Button className="mb-4" variant="outline-primary">
```
Comment (andyjih) [inline pages/docs/config.mdx]: I think this button is missing a link.

File: pages/docs/config/apps/community.mdx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+title: "Apps (Community)"
+date: "2021-02-11"
+pullQuote: "Apps are how Grouparoo can connect to any number of different tools such as your databases, data warehouses, and other 3rd party tools."
+breadcrumb_label: "Community Guide"
+---
+
+<Alert variant="primary">
+  <p className="mb-0">
+    This guide is covers configuration with Grouparoo's Code Config, which is
```
Comment (andyjih) [inline pages/docs/config/apps/community.mdx]: "This guide is covers" 
->
"This guide covers"

File: pages/docs/config/apps/community.mdx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    </a> instead.
+  </p>
+</Alert>
+
+<Alert variant="warning">
+  <p className="mb-0">
+    This guide assumes that you've{" "}
+    <a href="/docs/cli">installed the Grouparoo CLI</a> and used the CLI to{" "}
+    <a href="/docs/installation">
+      initialized the Grouparoo application locally
```
Comment (andyjih) [inline pages/docs/config/apps/community.mdx]: "initialized"
->
"initialize"

File: pages/docs/config/apps/enterprise.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
+---
+title: "Apps (Enterprise)"
+date: "2021-02-11"
+pullQuote: "Learn how to configure Apps in Grouparoo's Enterprise Edition."
+breadcrumb_label: "Enterprise Guide"
+---
+
+<Alert variant="primary">
+  <p>
+    This guide is covers configuration with Grouparoo's Enterprise Edition. If
```
Comment (andyjih) [inline pages/docs/config/apps/enterprise.mdx]: "This guide is covers" 
-> 
"This guide covers"

**Review by andyjih — CHANGES_REQUESTED** [review]
Great work! I had some comments. I think the intro section to most of the configuration docs has the same typo of "This guide is covers" so I stopped commenting on every single file. 

Also per your Discrepancy in Steps comments, I think we can keep the UI config docs under the Configure section for now. Perhaps we can add some guidance at the top of each of those UI config pages that they should run & deploy Grouparoo first?

File: pages/docs/config.mdx [hunk 5]
```diff
@@ -1,56 +1,41 @@
...
 
-- [Installing Grouparoo](/docs/deployment)
-- [Setting up Grouparoo the first time](/docs/config/first-time-setup)
-- [Adding Apps](/docs/config/apps)
-- [Adding Sources](/docs/config/sources)
-- [Adding Destinations](/docs/config/destinations)
-- [Defining a Profile Property Rule](/docs/config/properties)
-- [Creating Groups](/docs/config/groups)
+- [Intro to UI Config](/docs/config/ui-config)
+- [Adding Apps](/docs/config/apps/enterprise)
```
Comment (andyjih) [inline pages/docs/config.mdx:35]: Per your "Discrepancy in Steps" note, I wonder if we add some notes here about Running and Deploying before using the UI to configure.

File: data/docs-redirects.js [hunk 6]
```diff
@@ -94,11 +94,25 @@ const communityRedirects = [
...
+const configRedirects = [
+  { src: "/docs/config/apps", dest: "/docs/config/apps/community" },
+  { src: "/docs/config/sources", dest: "/docs/config/sources/community" },
+  { src: "/docs/config/properties", dest: "/docs/config/properties/community" },
+  { src: "/docs/config/groups", dest: "/docs/config/groups/community" },
+  {
+    src: "/docs/config/destinations",
+    dest: "/docs/config/destinations/community",
+  },
+];
```
Comment (evantahler) [inline data/docs-redirects.js:108]: Nice!

File: pages/docs/config.mdx (hunk 1 above)
Comment (evantahler) [reply pages/docs/config.mdx]: Also... it seems odd to have both a button and a link to the same page right near each other.

File: pages/docs/config/apps/community.mdx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+breadcrumb_label: "Community Guide"
+---
+
+<Alert variant="primary">
+  <p className="mb-0">
+    This guide is covers configuration with Grouparoo's Code Config, which is
+    the recommended way to configure your application. If you have{" "}
+    <a href="/meet">purchased the Enterprise Edition</a>, you may wish to
+    <a href="/docs/config/apps/enterprise">
+      use the Enterprise Apps Guide
```
Comment (evantahler) [inline pages/docs/config/apps/community.mdx]: There's a missing space on all the config pages `you may wishto use`

Comment (evantahler) [reply pages/docs/config/apps/community.mdx]: Another spot for a shared component maybe?

File: pages/docs/support/permissions.mdx [hunk 8]
```diff
@@ -1,7 +1,6 @@
 ---
```
Comment (evantahler) [inline pages/docs/support/permissions.mdx:1]: I don't think `docs/support/permissions` is just for enterprise users.  There are still teams and teamMembers in the community edition, and API keys.   It's OK to leave in the support section, but I'd remove the "Enterprise" title-tag

**Review by evantahler — COMMENTED** [review]
Awesome work with a lot of great updates! 

* Cleaning Up Frontmatter 👍 
* Nav always highlights parent 👍 
* The overall changes 👍

General Comments:

* We talked a bit about how to toggle between code-config and ui-config.  I think the choice you made to always default to code-config is good, and highlights the path we think most people will be taking
* None of the code-config examples show an actual code-config JS file.  Should they?
* On almost all of the pages you have the `Validating and Applying your Config` section - make that a shared component!  Also, perhaps it's a card or some other style so it stands apart.

**Comment by seancdavis** [comment]
Lots of updates here:

- Looks like @andyjih fixed all my stupid typos. (Slow down, Sean!) Thanks, Andy!
- Added a handful of shared components.
- Added code examples to each of the community pages.
- Removed "Enterprise" from permissions page. That was lingering from a previous story.
- Fixed the "Deployment" title, too.

**Pull request merged**